src/components/UserIncomes.tsx

```diff
@@ -0,0 +1,179 @@
+import { useState } from 'react';
+import { IPayment } from '../types';
+import Loading from './Loading';
+import { saveAs } from 'file-saver';
+import { renderTokenAmount } from '../utils/conversion';
+import { formatStringCompleteDate } from '../utils/dates';
+
+function UserIncomes({ payments }: { payments: IPayment[] }) {
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(50);
+
+  if (!payments || payments.length === 0) {
+    return <Loading />;
+  }
+
+  const filteredPayments = payments.filter(payment => {
```

Comment:
Several problems by doing filtering frontend side: 
- the page work only for 100 payments (default limit on the graph), after it will not display the other data
- so if you choose 200 per page you will never have it
- perf: you query all payments to potentially display 1

also it doesn't follow coding best practices as we handle to much logical here in data management.

Todo: 
- we need to move filtering in the graph query (pagination + date)
- we need a hook to get the payments paginated directly to centralized the logic and be easily reused